### ISSUE TITLE
Spack provided python does not provide Tkinter

### ISSUE THREAD

**Issue by KineticTheory** [LLNL/spack#1024]
Tkinter is normally a default part of python. But, if the python installer in spack doesn't find Tk/Tcl in system locations, this module will be broken. To enable this module, I believe we can add dependencies on Tk and Tcl and configure arguments for python

```
depends_on("tk")
depends_on("tcl")
...
"--with-tcltk-includes=-I%s/include -I%s/include" % ( spec['tk'].prefix, spec['tcl'].prefix ),
"--with-tcltk-libs=%s/lib %s/lib" % ( spec['tk'].prefix, spec['tcl'].prefix )
```

I also needed to set `TCL_LIBRARY` and `TK_LIBRARY` in my environment, but it seems there should be a better way to point python to the Tk/tcl libraries.

**Comment by mathstuf** [comment]
I think this should probably be a `+tk` variant on Python. It's such a rarely used module (IME) that defaulting it to on and causing builds of Tcl for everyone would be more painful than just finding those packages which need the module.

**Comment by adamjstewart** [comment]
This can be closed now that #1226 has been merged.

**Comment by KineticTheory** [comment]
I tested #1226 for my use case. Everything looks good..
